### PR TITLE
[FE+BE] Automate simple registration on dev and test

### PR DESCRIPTION
**Pull request by amurKontur**
## Summary by CodeRabbit

- **New Features**
  - Enhanced Keycloak integration with added methods for user registration, email verification, and token management.
  - Introduced new toolbar features for "smart city" users.

- **Bug Fixes**
  - Adjusted filtering logic in multiple tests to exclude specific projects.
  - Updated profile data retrieval methods to improve accuracy and reliability in tests.

- **Chores**
  - Added `@faker-js/faker` as a new development dependency.

- **Tests**
  - Improved end-to-end test scenarios by refining project filters, login flows, and toolbar text checks.
  - Introduced environment variables for Keycloak admin users in testing and development environments.

### CODE REVIEW

File: e2e/page-objects/keycloakPage.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    userObj.requiredActions = [];
+    const endpointToUpdateUser = `${domain}/admin/realms/${project.env === 'test' ? 'test' : 'dev'}/users/${id}`;
+    const updatedUserResponse = await apiContext.put(endpointToUpdateUser, {
+      data: userObj,
+      headers: {
+        Authorization: `Bearer ${newAdminToken}`,
+      },
+    });
+    expect(updatedUserResponse.status()).toEqual(204);
+  }
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/keycloakPage.ts]: no
it is autotest 
no need to write unit tests on autotests

File: e2e/login.spec.ts [hunk 2]
```diff
@@ -9,18 +9,12 @@ for (const project of projects) {
     pageManager,
   }) => {
     await pageManager.atBrowser.openProject(project);
-    // TO DO: remove this action after Atlas is launched
-    await pageManager.atBrowser.closeAtlasBanner(project);
     await pageManager.fromNavigationMenu.goToLoginPage();
     await pageManager.atLoginPage.typeLoginPasswordAndLogin(
       process.env.EMAIL!,
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/login.spec.ts:14]: without email test will fail and it is ok

File: e2e/registration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      username,
+    });
+
+    await pageManager.atBrowser.openProject(project, { skipCookieBanner: true });
+    await pageManager.fromNavigationMenu.goToLoginPage();
+    await pageManager.atLoginPage.typeLoginPasswordAndLogin(email, password, 1);
+
+    await pageManager.atProfilePage.checkLogoutBtnProfileTitleAndEmail(email);
+
+    const fullNameAfterRegistration = await pageManager.atProfilePage.getFullNameValue();
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/registration.spec.ts:69]: I have deleted it already, so no need

File: e2e/toolBarTextsWithUser.spec.ts [hunk 4]
```diff
@@ -23,21 +24,51 @@ const hiddenToolbarFeaturesUserWithNoRights = [
...
-    );
+
+    // TO DO: 7525 task will make map a feature and hidden for atlas. Adapt test once map is hidden
+    if (project.name !== 'atlas') {
+      await pageManager.atToolBar.checkTextsInToolbar(visibleTexts, hiddenTexts);
+    } else {
+      await expect(
+        await pageManager.atToolBar.getButtonByText('Measure distance'),
+      ).not.toBeVisible();
+    }
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/toolBarTextsWithUser.spec.ts:72]: it is already in-place

File: e2e/login.spec.ts [hunk 5]
```diff
@@ -9,18 +9,12 @@ for (const project of projects) {
...
       process.env.PASSWORD!,
       50,
     );
-    // TO DO: remove this action after Atlas is launched. Sometimes actions are too slow there, so 2 notification appears
-    try {
-      await pageManager.atBrowser.closeAtlasBanner(project);
-    } catch {}
     await pageManager.atProfilePage.checkLogoutBtnProfileTitleAndEmail(
       process.env.EMAIL!,
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/login.spec.ts:19]: test will fail then and it is okey

File: e2e/registration.spec.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      pageManager.atKeycloakPage.registerAndSeeVerificationEmailInfo(keycloakPage, {
+        fullName,
+        email,
+        password,
+      }),
+      pageManager.atKeycloakPage.getAdminToken({
+        project,
+        apiContext,
+        adminName: process.env.ADMIN_KEYCLOAK!,
+        adminPassword: process.env.ADMIN_KEYCLOAK_PASSWORD!,
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/registration.spec.ts:49]: test will fail then and it is okey

File: e2e/populationPopup.spec.ts [hunk 7]
```diff
@@ -16,9 +16,13 @@ type InputData = {
 };
 
 let projects = getProjects();
-// oam has no layers, smart-city has no population density layer
-// disaster-ninja temporally switched off untill 15482 issue is fixed
-projects = projects.filter((arg) => arg.name === 'atlas');
+// Oam has no layers, smart-city has no population density layer
+// Disaster-ninja temporally switched off untill 15482 issue is fixed
+// Atlas has no 'layers' feature for guest (and layers theirself)
+
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/populationPopup.spec.ts:22]: comments are ok

File: e2e/reloadWithUser.spec.ts [hunk 8]
```diff
@@ -15,11 +15,14 @@ for (const project of projects) {
...
+      const settingsValues = await pageManager.atProfilePage.getProfileData(project, {
+        shouldOsmEditorBeSeenOnAtlas: false,
+      });
       await pageManager.atProfilePage.compareUrlsAfterReload(project);
-      const settingValuesAfterReload =
-        await pageManager.atProfilePage.getAllProfileValuesAndStatuses();
+      const settingValuesAfterReload = await pageManager.atProfilePage.getProfileData(
+        project,
+        { shouldOsmEditorBeSeenOnAtlas: false },
+      );
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/reloadWithUser.spec.ts]: fixed

**Review by dqunbp — APPROVED** [review]

**Pull request merged**